assets/src/edit-story/elements/image/display.js

```diff
@@ -47,12 +68,28 @@ function ImageDisplay({ element, box }) {
     focalY
   );
 
+  useEffect(() => {
+    let timeout;
+    if (resourceList[resource.id]?.type !== 'fullsize') {
+      timeout = setTimeout(async () => {
+        await preloadImage(resource.src);
```

Comment:
I don't feel like this preload actually works here. It seems like the full-size image is inserted before fully loaded and loads while "onstage" (this is a big image with _Network throttling_ set to "Fast 3G"):

![Screen Shot 2020-08-26 at 20 01 50](https://user-images.githubusercontent.com/637548/91369337-85fb9b80-e7d9-11ea-8a73-f86771f77335.png)

Reply:
So, I've narrowed down the problem. It happens if the component re-renders for some reason. And it doesn't take much, just de-select and re-select it and it fails to recognize, that it hasn't yet preload the full image and tries to set it as the source, and you can the see the image slowly loading.

See this video showing an image loading correctly and one not being able to load before being re-selected:
https://youtu.be/QLP6cDuFgk4

Reply:
That helps a lot 👍 

Some of the issues that are now fixed were only happening for Unsplash images (no `resource.id` results in `id=undefined` duplication).
Other issues (e.g. loading two images instead of one) are about srcSet, for example, we cannot predict which image will be loaded by the browser (not to my knowledge), we can only read the `.currentSrc` after the fact - using that, I made another improvement in https://github.com/google/web-stories-wp/pull/3935/commits/393cb9b198cdbd7ca3d5040bed6858554548d847.

And as a bonus (I was looking at the network tab a lot) I made optimization to the layer thumbnail (we don't need to load high-quality image or use srcSet there) in https://github.com/google/web-stories-wp/pull/3935/commits/632757a9e9d2089edf6414d69e36f6c33a0b081c.

@barklund please re-test.
@diegovar PTAL